elfo-network/src/worker/flows_rx.rs

```diff
@@ -191,19 +200,93 @@ impl RxFlow<'_> {
             })
     }
 
-    pub(super) fn enqueue(self, envelope: Envelope, routed: bool) {
+    /// Enqueue response to the event queue. Returns `Err` if response must be
+    /// processed in real-time.
+    pub(super) fn try_enqueue_response(
+        self,
+        token: ResponseToken,
+        response: Result<Envelope, RequestError>,
+    ) -> Result<(), (ResponseToken, Result<Envelope, RequestError>)> {
+        let Some(queue) = self.flow.queue.as_mut() else {
+            return Err((token, response));
+        };
+
+        queue.push_back((RxFlowEvent::Response { token, response }, false));
+
+        Ok(())
+    }
+
+    /// Enqueue unbounded send. Returns `Err` if message must be processed in real-time.
+    pub(super) fn try_enqueue_unbounded(
+        self,
+        envelope: Envelope,
+        routed: bool,
+    ) -> Result<(), Envelope> {
+        let Some(queue) = self.flow.queue.as_mut() else {
+            return Err(envelope);
+        };
+
+        queue.push_back((
+            RxFlowEvent::Message {
+                envelope,
+                bounded: false,
+            },
+            routed,
+        ));
+
+        if routed {
+            self.flow.routed += 1;
+        }
+
+        Ok(())
+    }
+
+    /// Enqueue send. Returns `Err(..)` if there's no pusher.
+    pub(super) fn try_enqueue_send(
+        mut self,
+        envelope: Envelope,
+        routed: bool,
+    ) -> Result<(), Envelope> {
+        let Some(queue) = self.flow.queue.as_mut() else {
+            return Err(envelope);
+        };
+
+        queue.push_back((
+            RxFlowEvent::Message {
+                envelope,
+                bounded: true,
+            },
+            routed,
+        ));
+        self.acquire_direct(!routed);
```

Comment:
I don't understand what's happening with flow control in these methods and why it's different